src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -0,0 +1,196 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import difflib
+
+import azure.cli.core.telemetry as telemetry
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class CommandRecommender():
+    """Recommend a command for user when user's command fails.
+    It combines alladin recommendations and examples in help files."""
+
+    def __init__(self, command, parameters, extension, cli_ctx):
+        """
+        :param command: The command name in user's input, which should be corrected if misspelled.
+        :type command: str
+        :param parameters: The parameter arguments in users input.
+        :type parameters: list
+        :param extension: The extension name in user's input if the command comes from an extension.
+        :type extension: str
+        :param cli_ctx: CLI context when parser fails.
+        :type cli_ctx: knack.cli.CLI
+        """
+        self.command = command.strip()
+        self.extension = extension
+        self.cli_ctx = cli_ctx
+
+        self.parameters = self._normalize_parameters(parameters)
+        self.help_examples = []
+        self.aladdin_recommendations = []
+
+    def set_help_examples(self, examples):
+        """Set recommendations from help files"""
+
+        self.help_examples.extend(examples)
+
+    def _set_aladdin_recommendations(self):
+        """Set recommendations from aladdin service.
+        Call the aladdin service API, parse the response and set the recommendations.
+        """
+
+        import json
+        import requests
+        from requests import RequestException
+        from http import HTTPStatus
+        from azure.cli.core import __version__ as version
+
+        api_url = 'https://app.aladdin.microsoft.com/api/v1.0/suggestions'
+        headers = {'Content-Type': 'application/json'}
+        correlation_id = telemetry._session.correlation_id  # pylint: disable=protected-access
+        subscription_id = telemetry._get_azure_subscription_id()  # pylint: disable=protected-access
+
+        context = {
+            'versionNumber': version
+        }
```

Comment:
Add `errorType` field to here.

i.e.
```Python
context = {
    'versionNumber': version,
    'errorType': errorType
}
```
where error type is the user fault error type.

Reply:
Added